contracts/auction.sol

```diff
@@ -270,7 +270,12 @@ contract DutchAuction {
         }
 
         // Number of Rei = bid_wei / Rei = bid_wei / (wei_per_RDN * token_multiplier)
-        uint num = token_multiplier * bids[receiver_address] / final_price;
+        uint num = (token_multiplier * bids[receiver_address]) / final_price;
+
+        uint auction_tokens_balance = token.balanceOf(address(this));
```

Comment:
I think it can also work without casting to `address`

Reply:
For now, yes. But I used `address` because it is a dedicated API for the contract's address. I assumed that the probability that `address` disappears is smaller than the probability of `this` changing behavior.
Do you really want to use `this`?

Reply:
I don't think that this will ever change behaviour. Would break too many contracts. You can leave it as is if you want. Makes no difference in the final EVM code and readbility does not suffer, at least for me, with the typecast.